### PR TITLE
Add C++ solution to cses-1745

### PR DESCRIPTION
**Pull request by Cubik65536**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

------

Adding a C++ solution to the problem cses-1745 ([Money Sums](https://cses.fi/problemset/task/1745)), the code has been inspired by the Java version, and I used the same comment for the Java one.

### CODE REVIEW

File: solutions/cses-1745.mdx [hunk 1]
```diff
@@ -2,20 +2,84 @@
...
 
-## Solution
+``` cpp
+#include <bits/stdc++.h>
 
-**Time Complexity:** $\mathcal{O}(N\cdot X)$
+using namespace std;
 
-## Java Implementation
+using vi = vector<int>;
```
Comment (SansPapyrus683) [inline solutions/cses-1745.mdx:25]: don't think this is in the allowed macro list?

File: solutions/cses-1745.mdx [hunk 2]
```diff
@@ -2,20 +2,84 @@
...
+	}
+
+	cout << possible.size() << endl;
+
+	for (int sum : possible) {
+		cout << sum << " ";
+	}
+	cout << endl;
+
+	return 0;
```
Comment (SansPapyrus683) [inline solutions/cses-1745.mdx:76]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
bro this code is epic

File: solutions/cses-1745.mdx (hunk 1 above)
Comment (Cubik65536) [reply solutions/cses-1745.mdx:25]: This is written in the part [Type Aliases With `using`](https://usaco.guide/general/generic-code?lang=cpp#type-aliases-with-using) of USACO Guide, so I just used it, but remove it and use `vector<int>` in other parts shouldn't be a problem.

Comment (SansPapyrus683) [reply solutions/cses-1745.mdx:25]: address this & we're done

Comment (Cubik65536) [reply solutions/cses-1745.mdx]: > address this & we're done

done

**Review by SansPapyrus683 — APPROVED** [review]
wonderful!

File: solutions/cses-1745.mdx [hunk 3]
```diff
@@ -2,20 +2,80 @@
...
+			dp[i][current_sum] = dp[i - 1][current_sum];
+			int prev_sum = current_sum - x[i - 1];
+			if (prev_sum >= 0 && dp[i - 1][prev_sum]) {
+				dp[i][current_sum] = true;
+			}
+		}
+	}
+
+	// Check the number of possible sums.
+	vector<int> possible = vector<int>();
```
Comment (envyaims) [inline solutions/cses-1745.mdx:60]:
```suggestion
	vector<int> possible;
```
?

File: solutions/cses-1745.mdx [hunk 4]
```diff
@@ -2,20 +2,80 @@
...
-## Java Implementation
+	vector<int> x(n);
+	for (int i = 0; i < n; i++) {
+		cin >> x[i];
+	}
+
+	dp[0][0] = true;
+
+	/**
+	 * Loop through all coins and all possible sums.
```
Comment (envyaims) [inline solutions/cses-1745.mdx:43]: i think its better off having this in the explanation rather than rows of comments

**Review by envyaims — CHANGES_REQUESTED** [review]

File: solutions/cses-1745.mdx [hunk 5]
```diff
@@ -2,20 +2,80 @@
...
+	 * 1. If it is possible to make the sum with less than i coins,
+	 * Then DP[i][sum] will be true.
```
Comment (ryanchou-dev) [inline solutions/cses-1745.mdx:45]:
```suggestion
	 * then DP[i][sum] will be true.
```

File: solutions/cses-1745.mdx [hunk 6]
```diff
@@ -2,20 +2,80 @@
...
 
-## Java Implementation
+	vector<int> x(n);
+	for (int i = 0; i < n; i++) {
+		cin >> x[i];
+	}
+
+	dp[0][0] = true;
+
+	/**
```
Comment (ryanchou-dev) [inline solutions/cses-1745.mdx:42]:
```suggestion
	/*
```
?

File: solutions/cses-1745.mdx [hunk 7]
```diff
@@ -2,20 +2,80 @@
...
+bool dp[MAX_N + 1][MAX_SUM + 1];
+
+int main() {
+	int n;
+	cin >> n;
 
-## Java Implementation
+	vector<int> x(n);
+	for (int i = 0; i < n; i++) {
+		cin >> x[i];
```
Comment (ryanchou-dev) [inline solutions/cses-1745.mdx:37]: more descriptive than `x`?

File: solutions/cses-1745.mdx [hunk 8]
```diff
@@ -2,20 +2,80 @@
...
+
+	// Check the number of possible sums.
+	vector<int> possible = vector<int>();
+	for (int sum = 1; sum <= MAX_SUM; sum++) {
+		if (dp[n][sum]) {
+			possible.push_back(sum);
+		}
+	}
+
+	cout << possible.size() << endl;
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/cses-1745.mdx:67]: wha

File: solutions/cses-1745.mdx (hunk 4 above)
Comment (Cubik65536) [reply solutions/cses-1745.mdx]: I move it to the explanation part, but I didn't remove it from the Java version (which has the same comments), should I do that?

Comment (SansPapyrus683) [reply solutions/cses-1745.mdx]: yeah

**Review by envyaims — APPROVED** [review]
e

**Review by ryanchou-dev — APPROVED** [review]
✌️

**Comment by Cubik65536** [comment]
It seems that the master branch was not deployed because the yarn package service was down during the deployment, pls try to redeploy it if possible.
/cc @SansPapyrus683

**Comment by ryanchou-dev** [comment]
should be ok after another commit?

**Comment by Cubik65536** [comment]
yeah

**Pull request merged**